Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -79,6 +79,45 @@ final class CacheAdvanceTests: XCTestCase {
         XCTAssertEqual(messages, [])
     }
 
+    func test_isWritable_returnsTrueWhenStaticHeaderMetadataMatches() throws {
+        let originalCache = try createCache(overwritesOldMessages: false)
+        XCTAssertTrue(try originalCache.isWritable())
+
+        let sut = try createCache(overwritesOldMessages: false, zeroOutExistingFile: false)
+        XCTAssertTrue(try sut.isWritable())
+    }
+
+    func test_isWritable_throwsFileCorruptedWhenHeaderVersionDoesNotMatch() throws {
```

Comment:
`isWritable` needs to throw because we need to call `try setUpFileHandlesIfNecessary()` before reading the handle.

But I think you're right that we shouldn't be throwing `fileCorrupted` here. We should be returning `false` in this case.

Re whether we can swallow errors and not `throws` in this method... I could be convinced of that. The issue is if a client wants to know the difference between "version mismatch" and "the disk isn't readable", we'd be swallowing that.

Reply:
Looks like we `throw` in `synchronizeHeaderData()`. I need to think about how to unwind this 🤔

Reply:
> Re whether we can swallow errors and not `throws` in this method... I could be convinced of that. The issue is if a client wants to know the difference between "version mismatch" and "the disk isn't readable", we'd be swallowing that.

Yeah I'm torn on stuff like this. I do err on the side of more information. Also wondering if there are ways to throw less. But it's hard to do both 😓